### PR TITLE
[11.0][MIG] mrp_repair_refurbish

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Migration based on https://github.com/OCA/manufacture/pull/225. Added tests

### CODE REVIEW

**Review by HviorForgeFlow — APPROVED** [review]
LGTM :+1: Code and functional review

File: mrp_repair_refurbish/models/mrp_repair.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Copyright 2017-18 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class MrpRepair(models.Model):
+    _inherit = 'mrp.repair'
+
+    to_refurbish = fields.Boolean('To refurbish')
```
Comment (rousseldenis) [inline mrp_repair_refurbish/models/mrp_repair.py]: string not needed

File: mrp_repair_refurbish/models/mrp_repair.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            self.location_dest_id = self.refurbish_location_dest_id
+            self.refurbish_product_id = False
+            self.refurbish_location_dest_id = False
+
+    @api.multi
+    def action_repair_done(self):
+        res = super(MrpRepair, self).action_repair_done()
+        for repair in self:
+            if repair.to_refurbish:
+                move = self.env['stock.move'].create({
```
Comment (rousseldenis) [inline mrp_repair_refurbish/models/mrp_repair.py]: @lreficent Maybe externalizing the dict preparation in another function to help inheritance ?

File: mrp_repair_refurbish/models/product_template.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @api.depends('product_variant_ids',
+                 'product_variant_ids.refurbish_product_id')
+    def _compute_refurbish_product(self):
+        unique_variants = self.filtered(lambda template:
+                                        len(template.product_variant_ids) == 1)
+        for template in unique_variants:
+            template.refurbish_product_id = \
+                template.product_variant_ids.refurbish_product_id
+        for template in (self - unique_variants):
+            template.refurbish_product_id = False
```
Comment (rousseldenis) [inline mrp_repair_refurbish/models/product_template.py]: I think this is not needed as not stored (False is default value before compute)

File: mrp_repair_refurbish/models/product_template.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    @api.one
```
Comment (rousseldenis) [inline mrp_repair_refurbish/models/product_template.py]: This is deprecated

File: mrp_repair_refurbish/models/product_template.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    def _set_refurbish_product(self):
+        if len(self.product_variant_ids) == 1:
+            self.product_variant_ids.refurbish_product_id = \
+                self.refurbish_product_id
+
+    def _search_refurbish_product(self, operator, value):
+        products = self.env['product.product'].search([
```
Comment (rousseldenis) [inline mrp_repair_refurbish/models/product_template.py:41]: Just return
```suggestion
        return [('refurbish_product_id', operator, value)]
```
You avoid a search.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Some comments

File: mrp_repair_refurbish/models/product_template.py (hunk 5 above)
Comment (LoisRForgeFlow) [reply mrp_repair_refurbish/models/product_template.py:41]: it is compute non-stored field, how would that work?

File: mrp_repair_refurbish/models/mrp_repair.py (hunk 2 above)
Comment (LoisRForgeFlow) [reply mrp_repair_refurbish/models/mrp_repair.py]: sure, good point.

File: mrp_repair_refurbish/models/product_template.py (hunk 3 above)
Comment (LoisRForgeFlow) [reply mrp_repair_refurbish/models/product_template.py]: Indeed

File: mrp_repair_refurbish/models/product_template.py (hunk 5 above)
Comment (rousseldenis) [reply mrp_repair_refurbish/models/product_template.py:41]: Right! My bad

**Comment by LoisRForgeFlow** [comment]
@rousseldenis could you update your review?

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**